Yosemite/Yosemite/Stores/ShippingLabelStore.swift

```diff
@@ -0,0 +1,72 @@
+import Foundation
+import Networking
+import Storage
+
+/// Paper size options for printing a shipping label.
+public enum ShippingLabelPaperSize {
+    case label
+    case legal
+    case letter
+}
+
+/// Implements `ShippingLabelAction` actions
+///
+public final class ShippingLabelStore: Store {
+    private let remote: ShippingLabelRemoteProtocol
+
+    /// Shared private StorageType for use during then entire Orders sync process
+    ///
+    private lazy var sharedDerivedStorage: StorageType = {
+        storageManager.newDerivedStorage()
+    }()
+
+    public override init(dispatcher: Dispatcher, storageManager: StorageManagerType, network: Network) {
+        self.remote = ShippingLabelRemote(network: network)
+        super.init(dispatcher: dispatcher, storageManager: storageManager, network: network)
+    }
+
+    init(dispatcher: Dispatcher, storageManager: StorageManagerType, network: Network, remote: ShippingLabelRemoteProtocol) {
+        self.remote = remote
+        super.init(dispatcher: dispatcher, storageManager: storageManager, network: network)
+    }
+
+    /// Registers for supported Actions.
+    override public func registerSupportedActions(in dispatcher: Dispatcher) {
+        dispatcher.register(processor: self, for: ShippingLabelAction.self)
+    }
+
+    /// Receives and executes Actions.
+    override public func onAction(_ action: Action) {
+        guard let action = action as? ShippingLabelAction else {
+            assertionFailure("ShippingLabelStore received an unsupported action")
+            return
+        }
+
+        switch action {
+        case .printShippingLabel(let siteID, let shippingLabelID, let paperSize, let completion):
+            printShippingLabel(siteID: siteID, shippingLabelID: shippingLabelID, paperSize: paperSize, completion: completion)
+        }
+    }
+}
+
+private extension ShippingLabelStore {
+    func printShippingLabel(siteID: Int64,
+                            shippingLabelID: Int64,
+                            paperSize: ShippingLabelPaperSize,
+                            completion: @escaping (Result<ShippingLabelPrintData, Error>) -> Void) {
+        remote.printShippingLabel(siteID: siteID, shippingLabelID: shippingLabelID, paperSize: paperSize.rawValue, completion: completion)
+    }
+}
+
+private extension ShippingLabelPaperSize {
+    var rawValue: String {
+        switch self {
+        case .label:
+            return "label"
+        case .legal:
+            return "legal"
+        case .letter:
+            return "letter"
+        }
+    }
+}
```

Comment:
What do you think about removing this extension, and add `String` to the `ShippingLabelPaperSize` enum?

Like this:
```
public enum ShippingLabelPaperSize: String {
     case label
     case legal
     case letter
 }
```

Reply:
I replaced the enum here with the one from the networking layer in: https://github.com/woocommerce/woocommerce-ios/pull/3155/commits/7fd4a2596885cd00b700ab2465af51f05102663d

There I conformed the enum to `RawRepresentable` following other enums in the networking layer, and the string raw value is an internal detail.